sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -990,7 +990,14 @@ ur_program_handle_t ProgramManager::getBuiltURProgram(
     // emplace all subsets of the current set of devices into the cache.
     // Set of all devices is not included in the loop as it was already added
     // into the cache.
-    for (int Mask = 1; Mask < (1 << URDevicesSet.size()) - 1; ++Mask) {
+    int Mask = 1;
+    if (URDevicesSet.size() > sizeof(Mask) * 8 - 1) {
+      // Protection for the algorithm below. Although overflow is very unlikely
+      // to be reached.
```

Comment:
Sounds like it should be an `assert` then?

Reply:
discussed with Alexey in person, decided to make the message of exception more specific. 
done in https://github.com/intel/llvm/pull/16334/commits/637ad26731eb701dbe810dc9d2df9af1c24e8ace